javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -503,6 +519,52 @@ private static String expressionConvertToType(String name, MethodParameter conve
         }
     }
 
+    private static String expressionMultipartFormDataToBinaryData(String name, ClientModel model) {
+        // serialize model for multipart/form-data
+        StringBuilder builder = new StringBuilder().append("new MultipartFormDataHelper(requestOptions)");
+        Set<String> filePropertySerializedNames = new HashSet<>();
+        for (ClientModelProperty property : model.getProperties()) {
+            if (property.getWireType() == ClassType.BinaryData) {
+                // application/octet-stream
+                String serializedName = property.getSerializedName();
+                filePropertySerializedNames.add(serializedName);
+
+                // find corresponding filename property
+                String filenameExpression;
+                Optional<ClientModelProperty> filenameProperty = model.getProperties().stream()
+                        // here is a hack to find matching filename property by finding property of type String and of same serializedName
+                        .filter(p -> p.getWireType() == ClassType.String && Objects.equals(serializedName, p.getSerializedName()))
+                        .findFirst();
+                if (filenameProperty.isPresent()) {
+                    filenameExpression = name + "." + filenameProperty.get().getGetterName() + "()";
+                } else {
+                    filenameExpression = ClassType.String.defaultValueExpression(property.getSerializedName());
+                }
+
+                builder.append(String.format(
+                        ".serializeField(%1$s, %2$s.%3$s(), %4$s)",
+                        ClassType.String.defaultValueExpression(property.getSerializedName()),
+                        name,
+                        property.getGetterName(),
+                        filenameExpression
+                ));
+            } else if (filePropertySerializedNames.contains(property.getSerializedName())) {
+                // skip filename property
+            } else {
+                // text/plain
+                builder.append(String.format(
+                        ".serializeField(%1$s, %2$s.%3$s())",
+                        ClassType.String.defaultValueExpression(property.getSerializedName()),
+                        name,
+                        property.getGetterName()
+                ));
+            }
+            // TODO (weidxu): application/json
+        }
+        builder.append(".end().getRequestBody()");
+        return builder.toString();
+    }
```

Comment:
serialize a model of multipart/form-data to `BinaryData`, with the help of `MultipartFormDataHelper`